warehouse/router/upload.go

```diff
@@ -427,6 +431,37 @@ func (job *UploadJob) run() (err error) {
 	return nil
 }
 
+func (job *UploadJob) cleanupObjectStorageFiles() error {
+	destination := job.warehouse.Destination
+	cleanupObjectStorageFiles, _ := destination.Config["cleanupObjectStorageFiles"].(bool)
+	if cleanupObjectStorageFiles {
+		storageProvider := whutils.ObjectStorageType(destination.DestinationDefinition.Name, destination.Config, job.upload.UseRudderStorage)
+		fm, err := filemanager.New(&filemanager.Settings{Provider: storageProvider, Config: destination.Config})
+		if err != nil {
+			return err
+		}
+		stagingFileKeys := make([]string, len(job.stagingFiles))
+		for i, file := range job.stagingFiles {
+			stagingFileKeys[i] = fm.GetDownloadKeyFromFileLocation(file.Location)
+		}
+		if err = fm.Delete(job.ctx, stagingFileKeys); err != nil {
+			return err
+		}
+		loadingFiles, err := job.loadFilesRepo.GetByStagingFiles(job.ctx, job.stagingFileIDs)
+		if err != nil {
+			return err
+		}
+		loadingFileKeys := make([]string, len(loadingFiles))
+		for i, file := range loadingFiles {
+			loadingFileKeys[i] = fm.GetDownloadKeyFromFileLocation(file.Location)
+		}
+		if err = fm.Delete(job.ctx, loadingFileKeys); err != nil {
+			return err
+		}
```

Comment:
We can probably call the delete API once we have collected all the files simultaneously.
```suggestion
	loadingFiles, err := job.loadFilesRepo.GetByStagingFiles(job.ctx, job.stagingFileIDs)
	if err != nil {
		return fmt.Errorf("fetching loading files: %w", err)
	}

	fileToDelete := make([]string, len(loadingFiles)+len(job.stagingFiles))
	for i, file := range loadingFiles {
		fileToDelete[i] = fm.GetDownloadKeyFromFileLocation(file.Location)
	}
	for i, file := range job.stagingFiles {
		fileToDelete[i+len(loadingFiles)] = fm.GetDownloadKeyFromFileLocation(file.Location)
	}

	if err = fm.Delete(job.ctx, fileToDelete); err != nil {
		return fmt.Errorf("deleting files from object storage: %w", err)
	}
```

Reply:
Done